File: Python/week16/bubble_sort.py
from random import randint

lst = [randint(0, 99) for _ in range(99)]
R = range(len(lst) - 1)
print("Python Bubble Sort\n")
print(lst)
swap = True
while swap:
    swap = False
    for i in R:
        if lst[i] > lst[i + 1]:
            swap = True
            lst[i], lst[i + 1] = lst[i + 1], lst[i]
print()
print(lst)

File: Java/week16/BubbleSort.java

/*
 Filename: BubbleSort.java
 Author: Mikhail Zubko
 Created: 2025/05/01
 Purpose: 
*/
import java.util.Random;
import java.util.Arrays;

public class BubbleSort {
  public static void main(String[] args) {
    Random rand = new Random();
    int[] lst = new int[99];
    int len = lst.length - 1;
    // add random ints to the array
    for (int i = 0; i < lst.length; i++) {
      lst[i] = rand.nextInt(100); // upper bound is exclusive
    }
    System.out.println("Java Bubble Sort\n");
    System.out.println(Arrays.toString(lst));
    boolean swap = true;
    while (swap) {
      swap = false;
      for (int i = 0; i < len; i++) {
        if (lst[i] > lst[i + 1]) {
          swap = true;
          // Swap using temp var
          int temp = lst[i];
          lst[i] = lst[i + 1];
          lst[i + 1] = temp;
        }
      }
    }
    System.out.println();
    System.out.println(Arrays.toString(lst));
  }
}
